apps/server/test/lexical/server/code_intelligence/completion/translations/function_test.exs

```diff
@@ -12,6 +12,22 @@ defmodule Lexical.Server.CodeIntelligence.Completion.Translations.FunctionTest d
       assert [:deprecated] = completion.tags
     end
 
+    test "suggest arity 0 functions if not in a pipeline", %{project: project} do
+      {:ok, completion} =
+        project
+        |> complete("Application.loaded_app|")
+        |> fetch_completion(kind: :function)
+
+      assert completion.insert_text == "loaded_applications()"
+    end
+
+    test "do not suggest arity 0 functions if in a pipeline", %{project: project} do
+      {:error, :not_found} =
+        project
+        |> complete("|> Application.loaded_app|")
+        |> fetch_completion(kind: :function)
+    end
+
```

Comment:
something else is going on here; I don't consistently get the first parameters removed from pipes, yet the test on line 31 passes.

Reply:
I don't quite understand this question, do you mean this test: 

`    test "arity 1 omits arguments if in a pipeline", %{project: project} do`?

Reply:
after using the token stream, I think this issue you mentioned already fixed

Reply:
I created a function named `normalize_argument_names`, and "callable_snippet" and "label" both require context awareness, so passing "env" to them makes sense.